src/parse_instruction.py

```diff
@@ -287,13 +295,15 @@ def derived(
         return Instruction(mnemonic, args, replace(old.meta, synthetic=True))
 
     def __str__(self) -> str:
+        if not self.args:
+            return self.mnemonic
         args = ", ".join(str(arg) for arg in self.args)
         return f"{self.mnemonic} {args}"
 
 
 class ArchAsm(abc.ABC):
```

Comment:
would be nice to move all these classes closer to the top of the file. Good style since they are externally referenced and avoids some quoted type annotations